Doc/glossary.rst

```diff
@@ -363,6 +400,24 @@ Glossary
       the :term:`cyclic garbage collector <garbage collection>` is to identify these groups and break the reference
       cycles so that the memory can be reclaimed.
 
+   data race
+      A situation where multiple threads access the same memory location
+      concurrently, at least one of the accesses is a write, and the threads
+      do not use any synchronization to control their access.  Data races
+      lead to :term:`non-deterministic` behavior and can cause data corruption.
+      Proper use of :term:`locks <lock>` and other :term:`synchronization primitives
+      <synchronization primitive>` prevents data races.  See also
+      :term:`race condition` and :term:`thread-safe`.
```

Comment:
This should explain that data races are only possible via extensions and not via pure-python code.

Reply:
How about "Note that data races can only happen in native code, but that native code might be exposed in a Python API"?

Reply:
Done.